pulp_docker/app/tasks/tag.py

```diff
@@ -0,0 +1,39 @@
+from pulpcore.plugin.models import Repository, RepositoryVersion, ContentArtifact
+from pulp_docker.app.models import ManifestTag, Manifest
+
+
+def create_new_repository_version(manifest_pk, tag, repository_pk):
+    """
+    Create a new repository version out of the passed tag name and the manifest.
+
+    If the tag name is already associated with an existing manifest with the same digest,
+    no new content is created. Note that a same tag name cannot be used for two different
+    manifests. Due to this fact, an old ManifestTag object is going to be removed from
+    a new repository version when a manifest contains a digest which is not equal to the
+    digest passed with POST request.
+    """
+    manifest = Manifest.objects.get(pk=manifest_pk)
+    artifact = manifest._artifacts.all()[0]
+
+    existing_manifest_tag_list = ManifestTag.objects.filter(name=tag).exclude(
```

Comment:
s/existing_manifest_tag_list/existing_manifest_tag

Reply:
The method filter() returns Query Set what is some kind of list. Wouldn't it just obfuscate what the does?

Reply:
i see what you're saying, i thought this is about tag of a manifest list.
maybe coming up with a more generic name like existing_tag and new_tag will be more transparent?

Reply:
i also think that this query would return you existing tags which are part of other repos also. we need to limit so it queries within the given repo